miner/scroll_worker.go

```diff
@@ -521,8 +523,7 @@ func (w *worker) tryCommitNewWork(now time.Time, parent common.Hash, reorgReason
 	}
 
 	// check if we are reorging
-	reorging := w.chain.GetBlockByNumber(w.current.header.Number.Uint64()) != nil
```

Comment:
The problem was that `reorging` ended up being `false` for the block that we needed to add to generate the longest chain if it was created due to a circuit overflow. If that block didn't end up overflowing the circuit it would've force committed by this line [here](https://github.com/scroll-tech/go-ethereum/pull/1049/files#diff-1825330f8f75827945cb0144a6ab3dc034a3081f48ee96b0a318c5de9950832eL1050)